### PR TITLE
By country report

### PR DESCRIPTION
**Pull request by urfolomeus**
This was a (so far) one-off report to show the uptake of the register by country.
It's a little rough but has been committed in case we need it again.

### CODE REVIEW

**Comment by JiggyPete** [comment]
:+1: Good to keep this incase we need to do it again

File: lib/reports/by_country_report.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      @lookup[postcode_slug_for(adviser.firm)]
+    end
+    generate_report('Advisers', grouped_advisers, advisers.count)
+  end
+
+  def error_report
+    grouped_firms = @registered_firms.group_by do |firm|
+      @lookup[postcode_slug_for(firm)]
+    end
+    firms_with_errors = grouped_firms.select { |k, _| k.blank? }.values.flatten.map(&:id).uniq
```
Comment (asmega) [inline lib/reports/by_country_report.rb]: why do these firms have errors? i.e. why would `k` be blank?

Comment (urfolomeus) [inline lib/reports/by_country_report.rb]: `k` will be blank for any postcodes which return a blank country (i.e. those that are not found on the postcode lookup).

File: .rubocop.yml [hunk 2]
```diff
@@ -32,6 +32,7 @@ AllCops:
   - 'bin/**/*'
   - 'config/**/*'
   - 'db/schema.rb'
+  - 'lib/reports/by_country_report.rb'
```
Comment (munckymagik) [inline .rubocop.yml]: Why doesn't this file pass?

Comment (urfolomeus) [inline .rubocop.yml]: Length of the `fetch_countries` method. Peter and I tried to get it down below 15 lines yesterday and managed to get 1 line off by making it unreadable. :)

Decided it wasn't worth sinking more time into it and ignored the file instead.

Comment (urfolomeus) [inline .rubocop.yml]: Actually I'm going to have another go whilst I finish my lunch.

Comment (urfolomeus) [inline .rubocop.yml]: OK @munckymagik, this should be (a bit) better now.

**Comment by munckymagik** [comment]
:+1:

**Comment by asmega** [comment]
:+1:

**Pull request merged**